service/src/test/java/bio/terra/workspace/service/workspace/WorkspaceServiceTest.java

```diff
@@ -247,23 +264,43 @@ void testUpdateWorkspace() {
     String otherDescription = "The deprecated workspace";
 
     Workspace secondUpdatedWorkspace =
-        workspaceService.updateWorkspace(USER_REQUEST, workspaceId, null, otherDescription, null);
+        workspaceService.updateWorkspace(USER_REQUEST, workspaceId, null, null, otherDescription, null);
 
     // Since name is null, leave it alone. Description should be updated.
+    assertEquals(userFacingId, secondUpdatedWorkspace.getUserFacingId().orElse(null));
     assertEquals(name, secondUpdatedWorkspace.getDisplayName().orElse(null));
     assertEquals(otherDescription, secondUpdatedWorkspace.getDescription().orElse(null));
     assertEquals(propertyMap2, updatedWorkspace.getProperties());
 
     // Sending through empty strings and an empty map clears the values.
     Map<String, String> propertyMap3 = new HashMap<>();
     Workspace thirdUpdatedWorkspace =
-        workspaceService.updateWorkspace(USER_REQUEST, workspaceId, "", "", propertyMap3);
+        workspaceService.updateWorkspace(USER_REQUEST, workspaceId, "", "", "", propertyMap3);
+    assertEquals("", thirdUpdatedWorkspace.getUserFacingId().orElse(null));
     assertEquals("", thirdUpdatedWorkspace.getDisplayName().orElse(null));
     assertEquals("", thirdUpdatedWorkspace.getDescription().orElse(null));
 
     assertThrows(
         MissingRequiredFieldException.class,
-        () -> workspaceService.updateWorkspace(USER_REQUEST, workspaceId, null, null, null));
+        () -> workspaceService.updateWorkspace(USER_REQUEST, workspaceId, null, null, null, null));
+  }
+
+  @Test
+  void testUpdateWorkspaceUserFacingIdAlreadyExistsRejected() {
```

Comment:
I thought you already tested this earlier.

Reply:
`duplicateWorkspaceUserFacingIdRequestsRejected()` tests `createWorkspace()`, this tests `updateWorkspace()`. Both `createWorkspace()` and `updateWorkspace()` have validation logic.